builtin/logical/aws/path_static_roles.go

```diff
@@ -256,22 +260,32 @@ func (b *backend) validateRoleName(name string) error {
 	return nil
 }
 
-func (b *backend) validateIAMUserExists(ctx context.Context, req *logical.Request, username string) error {
+// validateIAMUser checks the user information we have for the role against the information on AWS. On a create, it uses the username
+// to retrieve the user information and _sets_ the userID. On update, it validates the userID and username.
+func (b *backend) validateIAMUserExists(ctx context.Context, req *logical.Request, entry *staticRoleEntry, isCreate bool) error {
 	c, err := b.clientIAM(ctx, req.Storage)
 	if err != nil {
-		return fmt.Errorf("unable to validate username %q: %w", username, err)
+		return fmt.Errorf("unable to validate username %q: %w", entry.Username, err)
 	}
 
 	// we don't really care about the content of the result, just that it's not an error
 	out, err := c.GetUser(&iam.GetUserInput{
-		UserName: aws.String(username),
+		UserName: aws.String(entry.Username),
 	})
 	if err != nil || out.User == nil {
-		return fmt.Errorf("unable to validate username %q: %w", username, err)
+		return fmt.Errorf("unable to validate username %q: %w", entry.Username, err)
 	}
-	if *out.User.UserName != username {
-		return fmt.Errorf("AWS GetUser returned a username, but it didn't match: %q was requested, but %q was returned", username, *out.User.UserName)
+	if *out.User.UserName != entry.Username {
+		return fmt.Errorf("AWS GetUser returned a username, but it didn't match: %q was requested, but %q was returned", entry.Username, *out.User.UserName)
 	}
+
+	if !isCreate && *out.User.UserId != entry.ID {
+		return fmt.Errorf("AWS GetUser returned a user, but the ID did not match: %q was requested, but %q was returned", entry.ID, *out.User.UserId)
+	} else {
+		// set userid
```

Comment:
nit: I think we could improve this comment if we provided some info on why we are doing this.

Reply:
done!